### PR TITLE
doc: lots of new doc; fixes

### PR DESCRIPTION
**Pull request by frasertweedale**
Add a bunch of doc.  Pls review and see if anything is unclear / inconsistent / missing.

### CODE REVIEW

File: src/Data/MIME.hs [hunk 1]
```diff
@@ -142,16 +231,33 @@ caseInsensitive = iso CI.mk CI.original
...
+matchContentType wantType wantSubtype (ContentType gotType gotSubtype _) =
+  wantType == gotType && maybe True (== gotSubtype) wantSubtype
+
 -- | Are the type and subtype the same? (parameters are ignored)
+--
 ctEq :: ContentType -> ContentType -> Bool
-ctEq (ContentType typ1 sub1 _) (ContentType typ2 sub2 _) =
-  typ1 == typ2 && sub1 == sub2
+ctEq (ContentType typ1 sub1 _) = matchContentType typ1 (Just sub1)
+{-# DEPRECATED ctEq "Use 'matchContentType' instead" #-}
```
Comment (romanofski) [inline src/Data/MIME.hs]: Is this actually necessary? The library isn't released yet, so no real reason as to deprecate it, rather then just deleting it?

Comment (frasertweedale) [reply src/Data/MIME.hs]: well... I'll delete it in a later commit, after I change purebred.  I just didn't want to make a breaking change in a doc commit.

Comment (frasertweedale) [reply src/Data/MIME.hs]: Then again, I probably shouldn't have made *any* code changes, but writing doc sometimes reveals bad APIs so it was a drive-by :)

Comment (frasertweedale) [reply src/Data/MIME.hs]: Thanks for reviewing @romanofski !

**Pull request merged**